terraform/context_apply_test.go

```diff
@@ -10940,3 +10940,78 @@ func TestContext2Apply_ProviderMeta_refreshdata_setInvalid(t *testing.T) {
 		t.Errorf("Expected unsupported argument error, none received")
 	}
 }
+
+func TestContext2Apply_expandModuleVariables(t *testing.T) {
```

Comment:
Request, I'm okay if this is dropped or addressed separately in the "we should have more tests" tasks -- but when I was testing yesterday, I also tested invalid references ex. `module.mod1["b"]` to see what happens, and in our tests I'd like to make sure those error in some fashion

Reply:
I think we can do that in a separate PR, since it's likely to be covered in the rewriting of the evaluation of module references.